Doc(anta_runner): Added how-to guide to migrate from eos_validate_state to anta_runner (#6465)
aristanetworks/ansible-avd: docs/howto/migrate-to-anta-runner.md

<!--
  ~ Copyright (c) 2025-2026 Arista Networks, Inc.
  ~ Use of this source code is governed by the Apache License 2.0
  ~ that can be found in the LICENSE file.
  -->

> # Migrating from eos_validate_state to anta_runner

Reviewer: I would keep the same title as navigation menu. 
Also I recommend "Migrate to anta_runner" as the menu option

```suggestion
# Migrate to anta_runner
```

Reviewer 2: <img width="254" height="313" alt="Screenshot 2026-02-03 at 21 10 39" src="https://github.com/user-attachments/assets/55c1a47e-5637-4556-97fd-2b1ba154ad9b" />
@joelbreton2 should we rename the one for cv_deploy?

Reviewer: I think so

Author: Done.

This guide helps you migrate from the deprecated `arista.avd.eos_validate_state` role to the new `arista.avd.anta_runner` role.

!!! note
    The `eos_validate_state` role was deprecated in AVD 5.7 and removed in AVD 6.0.0. The `anta_runner` role provides the same functionality with additional features and improvements.

For full documentation on the `anta_runner` role, see the [anta_runner documentation](../../ansible_collections/arista/avd/roles/anta_runner/README.md).

## Quick Start

> The simplest migration requires only updating the role name in your playbook:

Reviewer: ```suggestion

To use the new role, update the role name in your playbook. It is a good idea to also update the description.
```

Author: Partially applied your suggestion. Please check.

```diff
  ---
  - name: Validate Network State
    hosts: FABRIC
    connection: local
    gather_facts: false
    tasks:

>       - name: Validate states on EOS devices

Reviewer: ```suggestion
      - name: Run ANTA on EOS devices
```

Reviewer: this is the task name in the User Manual for the anta_runner role

Author: Done.

        import_role:
-         name: arista.avd.eos_validate_state
+         name: arista.avd.anta_runner
```

> This basic migration uses sensible defaults and will work for most use cases. However, there are some behavioral differences between the two roles. Review the sections below to understand these differences and how to achieve the same behavior as `eos_validate_state`.

Reviewer: ```suggestion
This example updates the role name to the new model. However, because there are functional differences between the roles, you must adjust your configuration to maintain the behavior of `eos_validate_state`. Review the sections below to identify the specific changes required for your environment.
```

Author: Reworded a bit. Please check.

## Connection Variables

> The `anta_runner` role supports both standard Ansible connection variables and ANTA-specific variables. The ANTA-specific variables take precedence when both are defined:

Reviewer: ```suggestion
The `anta_runner` role supports both standard Ansible connection variables and ANTA-specific variables. The ANTA-specific variables take precedence if both are defined:
```

Author: Fixed.

> The `anta_runner` role supports both standard Ansible connection variables and ANTA-specific variables. The ANTA-specific variables take precedence when both are defined:

Reviewer: This is the first topic discussed in the migration guide, and no changes are required. Do we want to move it later in the doc or remove it completly

Author: I decided to remove this section since no changes are required.

| Standard Ansible Variable | ANTA Variable (takes precedence) |
| :--- | :--- |
| `ansible_user` | `anta_user` |

> | `ansible_password` | `anta_password` |

Reviewer 3: Is it worth mentioning `ansible_httpapi_pass` & `ansible_httpapi_password`?

Author: I decided to remove that section completely since no changes are needed.

| `ansible_become` | `anta_enable` |
| `ansible_become_password` | `anta_enable_password` |
| `ansible_httpapi_use_ssl` | `anta_use_ssl` |
| `ansible_httpapi_port` | `anta_port` |

If you were using standard Ansible connection variables with `eos_validate_state`, they will continue to work with `anta_runner`. For more details, see [Connection Options](../../ansible_collections/arista/avd/roles/anta_runner/README.md#connection-options).

## Directory Structure Changes

> 

Reviewer: add a description of the changes, not sure if we need the flolders in quote

```suggestion
In the legacy `eos_validate_state` model, files were split between `intended`` and `reports`. In anta_runner, everything related to the test engine is consolidated under a single `anta` directory.
```

Author: Done.

    │   ├── structured_configs/
    │   └── test_catalogs/          # Optional: saved catalogs (save_catalog: true)
    ├── custom_anta_catalogs/       # Custom catalogs
    └── reports/
        ├── test_results/           # JSON results per device
        ├── FABRIC-state.csv
        └── FABRIC-state.md
    ```

=== "anta_runner"

    ```text
    inventory/
    ├── intended/
    │   └── structured_configs/
    └── anta/
        ├── avd_catalogs/           # AVD-generated catalogs per device (always saved)
        ├── user_catalogs/          # User-defined catalogs (previously called custom catalogs)
        └── reports/
            ├── anta_report.json
            ├── anta_report.csv
            └── anta_report.md
    ```

</div>

To customize directory and report paths in `anta_runner`:

```yaml
avd_catalogs_dir: "{{ inventory_dir }}/anta/avd_catalogs"
user_catalogs_dir: "{{ inventory_dir }}/anta/user_catalogs"
anta_reports_dir: "{{ inventory_dir }}/anta/reports"
anta_report_md_path: "{{ anta_reports_dir }}/anta_report.md"
anta_report_csv_path: "{{ anta_reports_dir }}/anta_report.csv"
anta_report_json_path: "{{ anta_reports_dir }}/anta_report.json"
```

## User-Defined Catalogs

> In `eos_validate_state`, custom ANTA catalogs were automatically loaded from the `custom_anta_catalogs_dir` directory. In `anta_runner`, the nomenclature has changed to *user-defined catalogs* and must be explicitly enabled:

Reviewer: ```suggestion
"In `eos_validate_state`, custom ANTA catalogs were automatically loaded from the `custom_anta_catalogs_dir` directory. In `anta_runner`, these are replaced by user-defined catalogs. This new model requires you to explicitly enable the catalogs in your configuration.
```

Author: Reworded this a bit. Please check.

            tags: [ leaf ]
    ```

    ```yaml title="inventory.yml"
    # Assign anta_tags to devices
    all:
      children:
        DC1_LEAFS:
          hosts:
            DC1-LEAF1A:
              anta_tags: [ leaf ]
            DC1-LEAF1B:
              anta_tags: [ leaf ]
    ```

</div>

For more details on tag-based filtering, see [Tag-Based Filtering](../../ansible_collections/arista/avd/roles/anta_runner/README.md#tag-based-filtering).

## Test Filtering

The `skip_tests` variable has been replaced with `avd_catalogs_filters`. The key difference is that `anta_runner` no longer uses AVD test categories (e.g., `AvdTestHardware`, `AvdTestNTP`). Instead, you specify ANTA test class names directly (e.g., `VerifyNTP`, `VerifyEnvironmentPower`).

<div class="grid" markdown>

=== "eos_validate_state"

    ```yaml
    skip_tests:
      - category: AvdTestHardware
      - category: AvdTestNTP
    ```

=== "anta_runner"

    ```yaml
    avd_catalogs_filters:
      - skip_tests:
          - VerifyEnvironmentPower
          - VerifyEnvironmentCooling
          - VerifyTemperature
          - VerifyTransceiversManufacturers
          - VerifyNTP
    ```

</div>

!!! note

>     The `anta_runner` role generates additional tests compared to `eos_validate_state`. If you want to match the exact test coverage of `eos_validate_state`, you can use `avd_catalogs_filters` to skip specific tests.

Reviewer: ```suggestion
     The anta_runner role provides expanded test coverage. If your workflow requires matching the exact legacy test suite, use the avd_catalogs_filters variable to exclude specific tests from the generated catalog.
```

Author: Simplified this. Please check.

For a complete list of available ANTA test names, see the [AVD-generated Catalog Test Index](../../ansible_collections/arista/avd/roles/anta_runner/README.md#avd-generated-catalog-test-index). For additional filtering options, see [Test-Based Filtering](../../ansible_collections/arista/avd/roles/anta_runner/README.md#test-based-filtering).

## Logging Verbosity

In `eos_validate_state`, logging verbosity was controlled by the `logging_level` variable.

In `anta_runner`, logging verbosity is controlled by **Ansible standard verbosity flags** (`-v`, `-vv`, `-vvv`, `-vvvv`):

<div class="grid" markdown>

=== "eos_validate_state"

    ```yaml
    logging_level: "DEBUG"  # Options: DEBUG, INFO, WARNING, ERROR, CRITICAL
    ```

=== "anta_runner"

    ```shell
    # Use Ansible verbosity flags instead
    ansible-playbook playbooks/fabric-validate.yaml -vvv
    ```

</div>

For more details, see [Logging and Troubleshooting](../../ansible_collections/arista/avd/roles/anta_runner/README.md#logging-and-troubleshooting).

> ## Test Catalog Storage

Reviewer: I recommend moving this section after the User-Defined Catalogs section

Author: Moved the section.

In `eos_validate_state`, saving test catalogs was optional and controlled by `save_catalog`. When enabled, the saved catalog included both AVD-generated tests and custom catalog tests merged together.

> In `anta_runner`, AVD-generated catalogs are **always** saved to the `avd_catalogs_dir` directory. Only the AVD-generated catalogs are saved, not the merged catalog that includes user-defined catalogs.

Reviewer: ```suggestion
In anta_runner, AVD-generated catalogs are saved to the avd_catalogs_dir. Note that only these auto-generated files are stored there; the final "merged" catalog—which combines AVD tests with your user-defined catalogs—exists only in memory during execution and is not saved as a standalone file.
```
AVD generated catalogs, merge catalogs and user-defined catalogs can be confusing to "new" ANTA users, maybe my understanding is wrong

Author: I tried to simplified this a bit. Please check.

```yaml
# Disable AVD-generated catalogs
avd_catalogs_enabled: false
# Enable user-defined catalogs only
user_catalogs_enabled: true
```

## Dry Run Mode

The `anta_runner` role does **not** support Ansible check mode (`--check`). To generate tests without executing them, use the `anta_runner_dry_run` variable instead:

<div class="grid" markdown>

=== "eos_validate_state"

    ```shell
    ansible-playbook playbooks/fabric-validate.yaml --check
    ```

=== "anta_runner"

    ```yaml
    anta_runner_dry_run: true
    ```

</div>

## Report Generation

### Showing Only Failed Tests

In `eos_validate_state`, the `only_failed_tests` variable controlled whether reports showed only failed tests.

In `anta_runner`, use `anta_report_exclude_statuses` to exclude specific test statuses from reports:

<div class="grid" markdown>

=== "eos_validate_state"

    ```yaml
    only_failed_tests: true
    ```

=== "anta_runner"

    ```yaml
    anta_report_exclude_statuses: [ success, skipped ]
    ```

</div>

### Disabling Report Generation

In `eos_validate_state`, CSV and Markdown report generations were controlled by boolean variables.

> In `anta_runner`, set the report path to `null` to disable generation of a specific report format:

Reviewer: in the example only the csv is set to null. Does this mean the md report can't be disabled ?

Author: Both can be disabled. I updated the example.

In `anta_runner`, this setting has moved to `eos_designs` and is configured through `validation_profiles`:

<div class="grid" markdown>

=== "eos_validate_state"

    ```yaml
    accepted_xcvr_manufacturers:
      - Arista Networks
      - Arastra, Inc.
      - Third Party Vendor
    ```

=== "anta_runner"

    ```yaml
    # eos_designs configuration
    validation_profiles:
      - name: datacenter
        hardware:
          transceiver_manufacturers:
            - Arista Networks
            - Arastra, Inc.
            - Third Party Vendor

    # Apply to node types
    spine:
      defaults:
        validation_profile: datacenter
    l3leaf:
      defaults:
        validation_profile: datacenter
    ```

</div>

For more details, see [Validation Profiles](../../ansible_collections/arista/avd/roles/anta_runner/README.md#validation-profiles).

## Fabric-Wide Validation

In `eos_validate_state`, fabric-wide validation tests (Loopback0 reachability, VTEP reachability, inband management reachability, DPS reachability, routing table entries for fabric underlay) were generated by default.

> In `anta_runner`, these tests are **disabled by default** and require enabling `avd_catalogs_extra_fabric_validation`.

Reviewer 3: We are already showing below `info` how to enable it to match original behavior.

```suggestion
In `anta_runner`, these tests are **disabled by default**.
```

Reviewer 3: The rest of the how-to looks good!

Author: Done. Thanks!

To get the same behavior as `eos_validate_state` (hardware tests generated and skipped by ANTA), you can use `custom_platform_settings`:

```yaml
custom_platform_settings:
  - platforms:
      - VEOS
      - VEOS-LAB
      - vEOS
      - vEOS-lab
      - CEOS
      - cEOS
      - ceos
      - cEOSLab
      - CloudEOS
    feature_support:
      hardware_validation: true
```

## Complete Migration Example

The following example shows a complete migration from `eos_validate_state` to `anta_runner` with equivalent behavior:

<div class="grid" markdown>

=== "eos_validate_state"

    ```yaml
    - name: Validate Network State
      hosts: FABRIC
      connection: local
      gather_facts: false
      tasks:
        - name: Validate states on EOS devices
          import_role:
            name: arista.avd.eos_validate_state
          vars:
            save_catalog: true
            only_failed_tests: true
    ```

=== "anta_runner"

    ```yaml
    - name: Validate Network State
      hosts: FABRIC
      connection: local
      gather_facts: false
      tasks:
        - name: Run ANTA on EOS devices
          import_role:
            name: arista.avd.anta_runner
          vars:

>             # Enable reachability tests (default in eos_validate_state)

Reviewer: ```suggestion
             avd_catalogs_extra_fabric_validation: true # (1)!
```
1. Enable reachability tests (default in eos_validate_state)

use annotation instead of inline comments, makes it easier to read and compare

Author: Done! Thanks for the suggestion, didn't know that.
